### DISCUSSION TITLE
Request tutorials about yarpmanager

### DISCUSSION THREAD

**Discussion by YANGGANG1971** [robotology/community#379] [Q&A] [answered]
On a physical robot, we used the yarpmanager to manage the files below. E.g. I want to find the original C++ files (uncompiled), what should I do?
 
![image](https://user-images.githubusercontent.com/55526431/72029884-b3d3b480-32ca-11ea-9c69-f039b91f4580.png)

In addition, I noted that on the right of the figure, there is an option called Parameters as follows. It is strange that on my computer, I cannot find the file called dragonfly2_config_left_bayer_640_480.ini and even cannot find the folder named camera. 
Where are these files? 
 
![image](https://user-images.githubusercontent.com/55526431/72029891-bafac280-32ca-11ea-9d53-9d95e602547c.png)

In addition, I try to use the terminal to configure settings. Type on the terminal.

> $ yarpview --from  camera/ dragonfly2_config_left_bayer_640_480.ini

but failed as follow. How do I configure these settings if I want to use the terminal?

![image](https://user-images.githubusercontent.com/55526431/72029898-c221d080-32ca-11ea-9493-c46f1d536398.png)

**Comment by Nicogene** [comment, marked as answer]
Hi @YANGGANG1971 ,

 `dragonfly2_config_left_bayer_640_480.ini`  and `ServerGrabberDualDragon.ini` are configuration files for running the dragonfly cameras of `iCub`.

In order to have it available, you have to clone and install `robots-configuration`. 
But the problem that I see here is that you are trying to launch the device from the laptop, it should be launched from the `icub-head` or `pc104`, because the dragonfly cameras are attached to the head of the robot.
In fact in `yarpmanager` application you can see that the "host" entry is `pc104`.

**Comment by YANGGANG1971** [comment]
My general question is how to find the original codes (uncompiled) through yarpmanager. 
To control the physical robot, some demos have been compiled and installed, shown above. 
I wanna know the detailed codes and change these demos for my own project.

**Comment by YANGGANG1971** [comment]
Also, learn to configure the settings using yarpmanager.

**Comment by Nicogene** [comment]
> My general question is how to find the original codes (uncompiled) through yarpmanager.

I don't know if I understood your question but you can't reach the source code through yarpmanager.

If you tell me which executable want to modify I can point you where it is the corresponding source code.
> Also, learn to configure the settings using yarpmanager.

You can find [here](http://yarp.it/yarpmanager.html) some information how to use `yarpmanager` and how it works. This documentation page is quite old, if it missing some information you need, let us know.

**Comment by YANGGANG1971** [comment]
I want to know that the tutorials about demoRedBall. 
(https://github.com/robotology/icub-basic-demos/tree/master/demoRedBall)
It is better to be close to this kind of tutorial below and it is easy to be understood. (http://www.icub.org/software_documentation/icub_hello_icub.html)

**Comment by pattacini** [comment]
Hi @YANGGANG1971
 
We don't have a tutorial on `demoRedBall` unfortunately and we don't have time right now to write it up.

**Comment by YANGGANG1971** [comment]
Got it. Thank you anyway.
In this yarpmanager tutorial (https://www.yarp.it/yarpmanager.html)

$ cd <path to YARP local repository>/data/yarpmanager/tests 
$ yarpmanager-console 
help 
     # a list of yarpmanager-console keywords 
list app 
     # you should see EyesViewer-Localhost, FakeEyes, ... 
 load app EyesViewer-Localhost 
 run 
 connect 
     # you should see moving red bar 


While connecting, errors occurred. 
![Screenshot from 2020-01-15 17-57-56](https://user-images.githubusercontent.com/55526431/72419813-e0069e00-37c0-11ea-9312-dfe47adc4a47.png)

**Comment by Nicogene** [comment]
You should type `run` followed by the ID of the application.
In this example it should be:

```
>> run  EyesViewer-Localhost
```

**Comment by Nicogene** [comment]
@YANGGANG1971 if you have not further need of help I will close this issue.
